webapp/src/jaas/components/conference/jaas_conference.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import {Client4} from 'mattermost-redux/client';
+import {id as pluginId} from '../../../manifest';
+import constants from '../../../constants';
+
+type Props = {}
+
+type State = {
+    jaasJwt?: string | null,
+    jaasRoom?: string | null,
+}
+
+const JWT = 'jwt';
+const JAAS_URL = '8x8.vc';
+
+export default class JaaSConference extends React.PureComponent<Props, State> {
+    api: any;
+    style = getStyle();
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {};
+    }
+
+    initJaaS(jwt: string, room: string) {
+        const ws = Math.max(document.documentElement.clientWidth ?? 0, window?.innerWidth ?? 0);
+        const hs = Math.max(document.documentElement.clientHeight ?? 0, window?.innerHeight ?? 0);
+        const options = {
+            width: ws,
+            height: hs,
+            roomName: room,
+            jwt,
+            parentNode: document.querySelector('#jitsiMeet')
+        };
+
+        this.api = new (window as any).JitsiMeetExternalAPI(JAAS_URL, options);
+    }
+
+    startJaaSMeetingWindow(jwt: string | null, meetingId: string | null) {
+        const options = {
+            method: 'post',
+            body: JSON.stringify({jaasJwt: jwt ?? '', jaasPath: meetingId}),
+            headers: {}
+        };
+
+        const baseUrl = `/plugins/${pluginId}`;
+        fetch(`${baseUrl}/api/v1/meetings/jaas/settings`, Client4.getOptions(options)).
+            then((result) => {
+                if (result.ok) {
+                    return result.json();
+                }
+
+                throw new Error(result.statusText);
+            }).
+            then((result) => {
+                this.setState({
+                    jaasRoom: result.jaasRoom,
+                    jaasJwt: result.jaasJwt
+                });
+            });
+    }
+
+    componentDidMount() {
+        if (!(window as any).JitsiMeetExternalAPI) {
+            // Check if the script is already loaded or not
+            if (document.getElementById(`${pluginId}_jitsi_meet_external_api_script`)) {
+                return;
+            }
+            const script = document.createElement('script');
+            script.type = 'text/javascript';
+            script.id = `${pluginId}_jitsi_meet_external_api_script`;
+            script.onload = () => {
+                const params = new URLSearchParams(window.location.search);
+                const jwt = params.get(JWT);
+                const meetingId = params.get(constants.MEETING_ID);
+
+                this.startJaaSMeetingWindow(jwt, meetingId);
```

Comment:
IIUC, there is a JWT token in the user's location URL, while in Mattermost? That seems unnecessary if that's the case

Reply:
Can you please elaborate more on this comment?

Reply:
@ayusht2810 What is in `window.location.search` at the time of this code running? Is this in a Mattermost webapp context? Meaning is `location.origin` the Mattermost server at this time?

Reply:
@mickmister Yes, the origin is the mattermost URL at this time, and the `location.search` contains the meeting ID and the JWT token.

Reply:
h JWT is being used to set the expiry of the meeting